pkg/application/registry/application/storage/rest.go

```diff
@@ -236,7 +236,7 @@ func (rs *REST) Update(ctx context.Context, name string, objInfo rest.UpdatedObj
 		}
 	}
 
-	if !reflect.DeepEqual(oldApp.Spec, app.Spec) && app.Status.Phase != applicationapi.AppPhaseRolledBack {
+	if !reflect.DeepEqual(oldApp.Spec, app.Spec) && oldApp.Status.Phase != applicationapi.AppPhaseRolledBack && oldApp.Status.Phase != applicationapi.AppPhaseTerminating {
```

Comment:
If old app status is rolledBack, we cannot change app status to another status, right? Is rooledBack status a "black hole" status?

Reply:
you are right, we will use app status not oldapp status.